arraylist.c

```diff
@@ -87,6 +87,23 @@ array_list_add(struct array_list *arr, void *data)
 	return array_list_put_idx(arr, arr->length, data);
 }
 
+/*
+ * Deleting the idx-th element in the array_list.
+ */
+void
+array_list_del_idx(struct array_list *arr, int idx)
+{
+	if (idx < 0 || idx >= arr->length) {
+		return;
+	}
+	if(arr->array[idx]) arr->free_fn(arr->array[idx]);
+	arr->length--;
+	for ( ; idx < arr->length; idx++) {
```

Comment:
wouldn't memmove be better?

Reply:
Indeed, it should have been... :)  Thanks!  Fixed.